src/EventStore.Core.Tests/Services/Transport/Tcp/TcpClientDispatcherTests.cs

```diff
@@ -0,0 +1,224 @@
+using EventStore.Core.Bus;
+using EventStore.Core.Services.Transport.Tcp;
+using NUnit.Framework;
+using System;
+using EventStore.Core.Authentication;
+using EventStore.Core.Messaging;
+using EventStore.Core.Tests.Authentication;
+using System.Linq;
+using EventStore.Core.Data;
+using EventStore.Core.Messages;
+using System.Text;
+using EventStore.Core.Services.UserManagement;
+using EventStore.Core.TransactionLog.LogRecords;
+using EventStore.Core.Services;
+
+namespace EventStore.Core.Tests.Services.Transport.Tcp
+{
+    [TestFixture]
+    public class TcpClientDispatcherTests
+    {
+        private readonly NoopEnvelope _envelope = new NoopEnvelope();
+        private const byte _version = (byte)ClientVersion.V1;
+
+        private ClientTcpDispatcher _dispatcher;
+        private TcpConnectionManager _connection;
+
+        [OneTimeSetUp]
+        public void Setup()
+        {
+            _dispatcher = new ClientTcpDispatcher();
+
+            var dummyConnection = new DummyTcpConnection();
+            _connection = new TcpConnectionManager(
+                Guid.NewGuid().ToString(), TcpServiceType.External, new ClientTcpDispatcher(),
+                InMemoryBus.CreateTest(), dummyConnection, InMemoryBus.CreateTest(), new InternalAuthenticationProvider(
+                                new Core.Helpers.IODispatcher(InMemoryBus.CreateTest(), new NoopEnvelope()), new StubPasswordHashAlgorithm(), 1),
+                TimeSpan.FromSeconds(10), TimeSpan.FromSeconds(10), (man, err) => { });
+        }
+
+        [Test]
+        public void when_unwrapping_message_that_does_not_have_version1_unwrapper_should_use_version2_unwrapper()
+        {
+            var dto = new TcpClientMessageDto.DeleteStream("test-stream", 2, true, false);
```

Comment:
We probably want to change this "2" here and use `ExpectedVersion.Any` for readability.